src/communication/network_interface_sr.c

```diff
@@ -11324,3 +11329,36 @@ smmon_get_server_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
   css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
 #endif // !WINDOWS
 }
+
+/*
+ * smmon_finalize_force - finalize memory_monitor forcely
+ *
+ * return:
+ *
+ *  rid(in):
+ *  request(in):
+ *  reqlen(in):
+ */
+void
+smmon_finalize_force (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  char *ptr;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int error = NO_ERROR;
+#if !defined(WINDOWS)
+  mmon_finalize ();
```

Comment:
What should be done if other thread accesses `mmon_Gl` during finalization? If a particular thread once checked that `mmon_Gl` is not nullptr and performs operations, a crash might occur.

Reply:
How about changing logic of mmon_finalize() like this?

temp_mmon_Gl = mmon_Gl;
mmon_Gl = nullptr;
// sleep a second
delete temp_mmon_Gl;

Reply:
> How about changing logic of mmon_finalize() like this?
> 
> temp_mmon_Gl = mmon_Gl; mmon_Gl = nullptr; // sleep a second delete temp_mmon_Gl;

It looks un-safe. The mmon_Gl is not volatile type. So, some thread use cached value after mmon_Gl is destroyed. And I think this approach can not guarantee that there are no threads in the momery monitor context.